core/state_processor.go

```diff
@@ -253,11 +253,8 @@ func ApplyStakingTransaction(
 
 	// Apply the transaction to the current state (included in the env)
 	gas, err = ApplyStakingMessage(vmenv, msg, gp, bc)
-	utils.Logger().Info().Msgf("ApplyStakingMessage: usedGas: %v, err: %v, stakingTxn:", gas, err)
-
-	// even there is error, we charge it
 	if err != nil {
-		return nil, gas, err
+		return nil, 0, err
```

Comment:
why gas is always 0? was the previous variable always 0 as well?

Reply:
The previous logic is wrong. We shouldn't charge gas when the staking txn failed.

Reply:
Okay wow this mistake lasted a long time.